app/models/nic/interface.rb

```diff
@@ -23,7 +23,10 @@ class Interface < Base
     delegate :network, :to => :subnet
 
     def vlanid
-      self.tag.blank? ? self.subnet.vlanid : self.tag
+      return self.tag unless self.tag.blank?
+      return self.subnet.vlanid if self.subnet
```

Comment:
I think you forgot a `.first` in the end of the first line. But then you actually do exactly the same as I suggested in the PR (but - and that's my opinion of course - less readable).

I do agree I can add a blank-check too, will do that when I figure out how the tests work.